src/section/mod.rs

```diff
@@ -0,0 +1,200 @@
+//! Helpers for adding custom sections to error reports
+use std::fmt::{self, Display, Write};
+
+pub mod help;
+
+#[non_exhaustive]
+#[derive(Debug)]
+pub(crate) enum Order {
+    AfterErrMsgs,
+    AfterBackTrace,
+    SkipEntirely,
+}
+
+/// A custom section for an error report.
+///
+/// # Details
+///
+/// Sections consist of two parts, a header, and an optional body. The header can contain any
+/// number of lines and has no indentation applied to it by default. The body can contain any
+/// number of lines and is always written after the header with indentation inserted before
+/// every line.
+///
+/// # Construction
+///
+/// Sections are meant to be constructed via `Into<Section>`, which is implemented for all types
+/// that implement `Display`. The constructed `Section` then takes ownership it the `Display` type
+/// and boxes it internally for use later when printing the report.
```

Comment:
"then takes ownership it the `Display` type" ~ I feel like maybe there's a typo here?

Reply:
it should be "of"